detoks/src/main/java/nl/tudelft/trustchain/detoks/db/DbHelper.kt

```diff
@@ -67,21 +77,31 @@ class DbHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null
         val collection = mutableListOf<Token>(token)
         new_row.put(COLUMN_SER_TOKEN, serialize(collection))
 
+        var table = TABLE_TOKEN
+        if (admin) {
+            table = TABLE_ADMIN_TOKEN
+        }
+
         val db = this.writableDatabase
 //        val cursor = db.rawQuery("SELECT * FROM $TABLE_NAME_TOKEN WHERE $COLUMN_TOKEN_ID=? OR $COLUMN_VALUE=?",
 //            arrayOf(token_id, value.toString()))
-        val newRowId = db.insert(TABLE_TOKEN, null, new_row) // null?
+        val newRowId = db.insert(table, null, new_row) // null?
         db.close()
         return newRowId
     }
 
     /**
      * Removes the specified token from the DB
      */
-    fun removeToken(token: Token) {
+    fun removeToken(token: Token, admin: Boolean = false) {
+        var table = TABLE_TOKEN
+        if (admin) {
+            table = TABLE_ADMIN_TOKEN
+        }
+
```

Comment:
I think the admin wallet should not remove any tokens.
It should keep all previous versions of the tokens, right?